examples/libs/algorand/asset-tx/main.ts

```diff
@@ -0,0 +1,91 @@
+import { DfnsWallet, singleSigTransactionToEncoded } from '@dfns/lib-algorand'
+import { DfnsApiClient } from '@dfns/sdk'
+import { AsymmetricKeySigner } from '@dfns/sdk-keysigner'
+import { Algodv2, encodeObj, makeAssetTransferTxnWithSuggestedParams, waitForConfirmation } from 'algosdk'
+
+import * as dotenv from 'dotenv'
+
+dotenv.config()
+
+const initDfnsWallet = async (walletId: string) => {
+  const signer = new AsymmetricKeySigner({
+    privateKey: process.env.DFNS_PRIVATE_KEY!,
+    credId: process.env.DFNS_CRED_ID!,
+    appOrigin: process.env.DFNS_APP_ORIGIN!,
+  })
+
+  const dfnsClient = new DfnsApiClient({
+    appId: process.env.DFNS_APP_ID!,
+    authToken: process.env.DFNS_AUTH_TOKEN!,
+    baseUrl: process.env.DFNS_API_URL!,
+    signer,
+  })
+
+  return DfnsWallet.init({
+    walletId: walletId,
+    dfnsClient,
+  })
+}
+
+const TESTNET_USDC_ASSET_ID = 10458941
+
+async function main() {
+  const receiverWallet = await initDfnsWallet(process.env.ALGORAND_RECEIVER_WALLET_ID!)
+  console.log('algorand receiver address: %s', receiverWallet.address)
+
+  const senderWallet = await initDfnsWallet(process.env.ALGORAND_SENDER_WALLET_ID!)
+  console.log('algorand sender address: %s', senderWallet.address)
+
+  const algod = new Algodv2(process.env.ALGORAND_NODE_API_KEY!, process.env.ALGORAND_NODE_URL!)
+
+  // check that our receiver wallet has opted in for the asset
+  // otherwise, we need to opt in. (The receiver wallet needs at least 0.1 ALGO) 
+  if (!await optInForAsset(algod, receiverWallet)) {
+    console.log("receiver didn't optin for the asset")
+    const txHash = await SendUsdcTransferTxn(algod, receiverWallet, receiverWallet, 0)
+    console.log(`optin transaction broadcasted: ${txHash}`)
+    await waitForConfirmation(algod, txHash, 4)
+    console.log("optin transaction confirmed")
+  }
+
+  // Send USDC
+  console.log("sending 1 USDC")
+  const txHash = await SendUsdcTransferTxn(algod, senderWallet, receiverWallet, 1000000)
+  console.log(`USDC transfer broadcasted: ${txHash}`)
```

Comment:
just an idea, would be nice to end the example with an optout transaction. both show casing how optout works and also make the example completely rerunnable because the next run, the receiver has to opt in again I assume.

Reply:
Yeah that's nice indeed. What do you think about using atomic transfers?

(https://developer.algorand.org/docs/get-details/atomic_transfers/#group-transactions)

Is it overkilled or a nice to have demonstration?